src/main/java/org/folio/services/invoice/PoLinePaymentStatusUpdateService.java

```diff
@@ -145,46 +149,39 @@ private Map<String, List<InvoiceLine>>  groupInvoiceLinesByPoLineId(List<Invoice
   /**
    * Retrieves PO Lines associated with invoice lines and calculates expected PO Line's payment status
    * @param poLineIdsWithInvoiceLines map where key is PO Line id and value is list of associated invoice lines
-   * @return map where key is {@link CompositePoLine} and value is expected PO Line's payment status
+   * @return map where key is {@link PoLine} and value is expected PO Line's payment status
    */
-  private Future<Map<CompositePoLine, CompositePoLine.PaymentStatus>> fetchPoLines(Map<String, List<InvoiceLine>> poLineIdsWithInvoiceLines,
-    RequestContext requestContext) {
-    List<Future<CompositePoLine>> futures = poLineIdsWithInvoiceLines.keySet()
-      .stream()
-      .map(poLine -> orderLineService.getPoLine(poLine, requestContext))
-      .toList();
-
-    return collectResultsOnSuccess(futures)
-      .map(compositePoLines ->  compositePoLines.stream()
-        .collect(Collectors.toMap(poLine -> poLine, poLine -> getPoLinePaymentStatus(poLineIdsWithInvoiceLines.get(poLine.getId())))));
+  private Future<Map<PoLine, PaymentStatus>> fetchPoLines(Map<String, List<InvoiceLine>> poLineIdsWithInvoiceLines,
+      RequestContext requestContext) {
+    List<String> poLineIds = poLineIdsWithInvoiceLines.keySet().stream().toList();
+    return orderLineService.getPoLinesByIds(poLineIds, requestContext)
```

Comment:
will speedup process of paying invocie with multiple lines even more, because I remember from previous test runs it was not very fast operation in case in invoice has 50, 100, 500 lines